sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -142,30 +142,70 @@ export class AppConfigurationClient {
     options: GetConfigurationSettingOptions = {}
   ): Promise<GetConfigurationSettingResponse> {
     return await this.spanner.trace("getConfigurationSetting", options, async (newOptions) => {
-      const response = (await this.client.getKeyValue(id.key, {
+      const originalResponse = await this.client.getKeyValue(id.key, {
         label: id.label,
         select: newOptions.fields,
         ...newOptions,
-        ...checkAndFormatIfAndIfNoneMatch(newOptions)
-      })) as GetConfigurationSettingResponse;
+        ...checkAndFormatIfAndIfNoneMatch(id, newOptions)
+      });
+
+      const response: GetConfigurationSettingResponse = {
+        ...originalResponse,
+        _response: originalResponse._response,
+        statusCode: originalResponse._response.status
+      };
 
       // 304 only comes back if the user has passed a conditional option in their
       // request _and_ the remote object has the same etag as what the user passed.
-      if (response._response.status === 304) {
-        throw new ResponseBodyNotFoundError(
-          "The requested setting's value has not changed since the last request.",
-          "Resource same as remote",
-          response._response.status,
-          response._response.request,
-          response._response,
-          null
-        );
+      if (response.statusCode === 304) {
+        return AppConfigurationClient.craftSpecial304Response(response);
       }
 
       return response;
     });
   }
 
+  private static craftSpecial304Response(originalResponse: GetConfigurationSettingResponse): GetConfigurationSettingResponse {
+    const newResponse: GetConfigurationSettingResponse = {
+        ...originalResponse,
+      _response: originalResponse._response
+    };
+
+    Object.defineProperty(newResponse, '_response', {
+      enumerable: false
+    });
+
+    // TODO:  can I identify these fields in a less manual manner?
+    const names: (keyof ConfigurationSetting)[] = [
```

Comment:
I'd almost prefer having a dummy `ConfigurationSetting` object you could `Object.keys` from.

Reply:
I was really hoping someone would say "use <function you've never heard of>" to fix this. I could create a dummy object but it seems like basically the same issue - I'd have to keep updating the instance of the object each time I added a new field (unless this is not what you meant).

I did type the array with a 'keyof' so I can't go out of bounds with the key names though.

Reply:
Yeah, except keyof just makes sure nothing extra gets added. It doesn't enforce that *all* keys are used, e.g.

```ts
interface Foo {
    A: number;
    B?: string;
    C: boolean;
}

// this is valid
const names: (keyof Foo)[] = [
    "A",
    "B"
]

// this is also valid
const names2: (keyof Foo)[] = [
    "A",
    "C"
]

// not valid, B is missing
const dummyObj: Required<Foo> =  {
    A: 42,
    B: "string"
}

// not valid, B is missing
const dummyObj: Required<Foo> =  {
    A: 42,
    C: true
}
```

Reply:
Yes, I've got a way to test that that I'm about to push. I think you're going to like it. :)

Reply:
My problem with the current implementation is that it's easy to forget to add a new key and then you're no longer consistent.

Reply:
Yes, I agree. I added a test - check out helpers.ts (`getAllConfigurationSettingFields`)